app/controllers/hyrax/dashboard/nest_collections_controller.rb

```diff
@@ -2,39 +2,42 @@ module Hyrax
   module Dashboard
     class NestCollectionsController < ApplicationController
       include Blacklight::Base
-      class_attribute :form_class
+      class_attribute :form_class, :new_collection_form_class
       self.form_class = Hyrax::Forms::Dashboard::NestCollectionForm
+      self.new_collection_form_class = Hyrax::Forms::CollectionForm
 
-      def new_within
-        @form = build_within_form
-      end
-
+      # Add this collection within another collection (this collection as child)
       def create_within
         @form = build_within_form
         if @form.save
-          notice = I18n.t('create_within', scope: 'hyrax.dashboard.nest_collections_form', child_title: @form.child.title, parent_title: @form.parent.title)
-          redirect_to dashboard_collection_path(@form.child), notice: notice
+          notice = I18n.t('create_within', scope: 'hyrax.dashboard.nest_collections_form', child_title: @form.child.title.first, parent_title: @form.parent.title.first)
+          redirect_to redirect_path(item: @form.child), notice: notice
         else
-          render 'new_within'
+          flash[:error] = @form.errors.full_messages
+          redirect_to redirect_path(item: @form.child)
         end
       end
 
-      def process_nesting
-        notice = "No mapping took place"
-        unless params[:parent_id] == "none"
-          parent_coll = Collection.find params[:parent_id]
-          child_coll = Collection.find params[:child_id]
-
-          child_coll.member_of_collections << parent_coll
-          child_coll.save!
-
-          notice = "Collections were mapped!"
+      # create and link new subcollection under this collection, (this collection as parent)
+      def new_under
```

Comment:
I'm confused what is happening here. My expectation with a method prefix "new_" is that the action would not be redirecting but instead rendering HTML or JSON. The "create_" action would handle redirecting on success or rendering the "new_" action's template on failure. The request cycle would look as follows:

![http request mermaid](https://user-images.githubusercontent.com/2130/33272639-bf755e2a-d358-11e7-8926-a902e459986e.png)